designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ScenarioActivityApiEndpoints.scala

```diff
@@ -0,0 +1,235 @@
+
+  import ScenarioActivityApiEndpoints.Dtos._
+  import TapirCodecs.ContentDispositionCodec._
+  import TapirCodecs.HeaderCodec._
+  import TapirCodecs.ScenarioNameCodec._
+  import TapirCodecs.VersionIdCodec._
+  import TapirCodecs.ErrorsCodecs._
+
+  private val scenarioNotFoundErrorOutput: EndpointOutput.OneOf[BusinessError, BusinessError] = oneOf[BusinessError](
+    oneOfVariantFromMatchType(
+      NotFound,
+      plainBody[BusinessError.ScenarioNotFoundError]
+        .example(
+          Example.of(
+            summary = Some("No scenario {scenarioName} found"),
+            value = BusinessError.ScenarioNotFoundError(ProcessName("'example scenario'"))
+          )
+        )
+    )
+  )
+
+  lazy val scenarioActivityEndpoint: SecuredEndpoint[ProcessName, BusinessError, ScenarioActivity, Any] =
+    baseNuApiEndpoint
+      .summary("Scenario activity service")
+      .tag("Scenario")
+      .get
+      .in("processes" / path[ProcessName]("scenarioName") / "activity")
+      .out(
+        statusCode(Ok).and(
+          jsonBody[ScenarioActivity].example(
+            Example.of(
+              summary = Some("Display scenario activity"),
+              value = ScenarioActivity(
+                comments = List(
+                  Comment(
+                    id = 1L,
+                    processVersionId = 1L,
+                    content = "some comment",
+                    user = "test",
+                    createDate = Instant.parse("2024-01-17T14:21:17Z")
+                  )
+                ),
+                attachments = List(
+                  Attachment(
+                    id = 1L,
+                    processVersionId = 1L,
+                    fileName = "some_file.txt",
+                    user = "test",
+                    createDate = Instant.parse("2024-01-17T14:21:17Z")
+                  )
+                )
+              )
+            )
+          )
+        )
+      )
+      .errorOut(scenarioNotFoundErrorOutput)
+      .withSecurity(auth)
+
+  lazy val addCommentEndpoint: SecuredEndpoint[AddCommentRequest, BusinessError, Unit, Any] =
+    baseNuApiEndpoint
+      .summary("Add scenario comment service")
+      .tag("Scenario")
+      .post
+      .in(
+        ("processes" / path[ProcessName]("scenarioName") / path[VersionId]("versionId") / "activity"
+          / "comments" / stringBody).mapTo[AddCommentRequest]
+      )
+      .out(statusCode(Ok))
+      .errorOut(scenarioNotFoundErrorOutput)
+      .withSecurity(auth)
+
+  lazy val deleteCommentEndpoint: SecuredEndpoint[DeleteCommentRequest, BusinessError, Unit, Any] =
+    baseNuApiEndpoint
+      .summary("Delete process comment service")
+      .tag("Scenario")
+      .delete
+      .in(
+        ("processes" / path[ProcessName]("scenarioName") / "activity" / "comments"
+          / path[Long]("commentId")).mapTo[DeleteCommentRequest]
+      )
+      .out(statusCode(Ok))
+      .errorOut(scenarioNotFoundErrorOutput)
```

Comment:
in `ScenarioActivitySpec` (shoudn't it be `ScenarioActivityAPISpec`?) there is a test that checks if the endpoint returns 500 in case of deleting non-existing comment. Shouldn't we add this error def here?

Reply:
Done